### PR TITLE
ML-366 ECLdoc Utility

### PR DESCRIPTION
**Pull request by RogerDev**
Initial Population
Clean up documentation
Add details on installation
Fix several bugs
Add Example for generating ML Documentation

Signed-off-by: Roger Dev <Roger.Dev@LexisNexisRisk.com>

### CODE REVIEW

File: testing/test/intest/inintest/eclcc.log [hunk 1]
```diff
@@ -0,0 +1,10 @@
+00000000 2017-06-04 17:24:03  8223  8223 Plugin /home/sarthak/runtime/opt/HPCCSystems/6.5.0/clienttools/plugins/libdmetaphone.so exports getECLPluginDefinition but does not export ECL - not loading
```
Comment (richardkchapman) [inline testing/test/intest/inintest/eclcc.log]: I don't think this file should be checked in

File: testing/test/intest/in1intest/eclcc.log [hunk 2]
```diff
@@ -0,0 +1,10 @@
+00000000 2017-06-04 17:24:03  8207  8207 Plugin /home/sarthak/runtime/opt/HPCCSystems/6.5.0/clienttools/plugins/libdmetaphone.so exports getECLPluginDefinition but does not export ECL - not loading
```
Comment (richardkchapman) [inline testing/test/intest/in1intest/eclcc.log]: I don't think this file should be checked in

File: testing/test/eclcc.log [hunk 3]
```diff
@@ -0,0 +1,11 @@
+00000000 2018-03-24 05:42:48  3695  3695 Loading plugin /opt/HPCCSystems/6.5.0/clienttools/plugins/libfileservices.so[lib_fileservices] version = FILESERVICES 2.1.4
```
Comment (richardkchapman) [inline testing/test/eclcc.log]: remove

File: testing/import-test/eclcc.log [hunk 4]
```diff
@@ -0,0 +1,10 @@
+00000000 2017-08-12 16:46:41 28305 28305 Plugin /home/sarthak/runtime/opt/HPCCSystems/6.5.0/clienttools/plugins/libdmetaphone.so exports getECLPluginDefinition but does not export ECL - not loading
```
Comment (richardkchapman) [inline testing/import-test/eclcc.log]: remove

File: testing/imp_config.cfg [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+[OUTPUT]
+root=/media/sarthak/Data/ecldoc/testing/imp-docs
+
+[ECLCC]
+-I../eclbundles/ml-bundles=1
+-I../test/intest=1
+
+[EXDOC]
+paths=/media/sarthak/Data/ecldoc/testing/eclbundles/ml-bundles-docs/xml,
```
Comment (richardkchapman) [inline testing/imp_config.cfg]: Absolute paths probably a bad idea...

File: src/ecldoc/parseDoc.py~ [hunk 6]
```diff
@@ -0,0 +1,196 @@
+import re
```
Comment (richardkchapman) [inline src/ecldoc/parseDoc.py~]: this looks like a backup file that should not be checked in

File: src/ecldoc/genXML.py~ [hunk 7]
```diff
@@ -0,0 +1,486 @@
+import os
```
Comment (richardkchapman) [inline src/ecldoc/genXML.py~]: This file should not be checked in

File: src/ecldoc/USERDOC.txt [hunk 8]
```diff
@@ -0,0 +1,3 @@
+1) Period to denote first line in Comment
```
Comment (richardkchapman) [inline src/ecldoc/USERDOC.txt]: This file seems a little short on content

File: src/ecldoc/Templates/html/css/simple-sidebar.css~ [hunk 9]
```diff
@@ -0,0 +1,252 @@
+ body {
```
Comment (richardkchapman) [inline src/ecldoc/Templates/html/css/simple-sidebar.css~]: Not sure this file should be present

File: src/ecldoc/Templates/html/content.tpl .html.orig [hunk 10]
```diff
@@ -0,0 +1,144 @@
+<!DOCTYPE html>
```
Comment (richardkchapman) [inline src/ecldoc/Templates/html/content.tpl .html.orig]: Not sure this file should be present

File: .gitignore [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+# OS generated files #
+######################
+.DS_Store
+.DS_Store?
+._*
+.Spotlight-V100
+.Trashes
+ehthumbs.db
+Thumbs.db
+
```
Comment (richardkchapman) [inline .gitignore]: perhaps you should have added *~ and eclcc.log and *.odt ?

Comment (RogerDev) [reply .gitignore]: I can get rid of eclcc.log.  That was a mistake.  Documentation.odt is the main documentation for the project.

**Pull request merged**